packages/ui-components/cypress/support/customPercyCommand.js

```diff
@@ -35,10 +25,125 @@ const installCustomPercyCommand = ({ before, elementOverrides } = {}) => {
       $el.css({ visibility: 'hidden' })
     })
 
-    // if we're in interactive mode via (cypress open)
-    // then bail immediately
+    // Use takeRecords to flush all pending mutations that have not yet
+    // been processed by the observer's handler prior to disconnecting.
+    this.mutationStack = [...(observer.takeRecords() || [])]
+
+    observer.disconnect()
+  }
+
+  resetOverrides () {
+    _.forEachRight(this.mutationStack, ({
+      type,
+      target,
+      attributeName,
+      oldValue,
+      addedNodes,
+      removedNodes,
+    }) => {
+      if (type === 'attributes') {
+        target.setAttribute(attributeName, oldValue)
+
+        return
+      }
+
+      if (type === 'childList') {
+        if (addedNodes.length) {
+          addedNodes.forEach((addedNode) => {
+            target.removeChild(addedNode)
+          })
+        }
+
+        if (removedNodes.length) {
+          removedNodes.forEach((removedNode) => {
+            target.insertBefore(removedNode, removedNode.previousSibling)
+          })
+        }
+
+        return
+      }
+
+      // TODO: support more mutations
+    })
+
+    this.mutationStack = undefined
+  }
+}
+
+const applySnapshotMutations = ({
+  log,
+  snapshotWidth,
+  snapshotElementOverrides,
+  defaultWidth,
+  defaultHeight,
+}) => {
+  let elementOverrideManager
+
+  if (Object.keys(snapshotElementOverrides).length) {
+    elementOverrideManager = new ElementOverrideManager()
+  }
+
+  return cy.viewport(snapshotWidth, defaultHeight, { log: false })
+  .then(() => {
+    if (log) {
+      Cypress.log({
```

Comment:
I think this ends up working out because the snapshot is taken asynchronously when the log is finished, but no reason not to make the intention clear. I'll create the log after the synchronous override logic and make sure the log creates the snapshot and then immediately ends.